### DISCUSSION TITLE
Automatic Locale Detection event?

### DISCUSSION THREAD

**Discussion by ArenoRobert** [vercel/next.js#22790] [Help] [answered]
Hi!
We are currently using the i18n cookie to set which currency we are using on our ecommerce site.
That works perfectly when the user is changing locale using a dropdown, but not when using Automatic Locale Detection of course. 

Is there some way I can hijack the automatic locale detection and trigger a state change anywhere else?

**Comment by z4nr34l** [comment, marked as answer]
Automatic locale detection is basing on `Accept-Language` header as can be found in docs https://nextjs.org/docs/advanced-features/i18n-routing. I think you can just use same method.

**Reply by ArenoRobert** [reply]
Ah, great.

Where would be suitable to mimic the exact same functionality? 
Just to be executed on index page and only the first time renedered (I guess?).

**Reply by z4nr34l** [reply]
IMHO I would prefer to do this in custom App (_app.js) as a global state for whole application, as it should be able to change prices anywhere in app

**Reply by ArenoRobert** [reply]
Yes, I've StoreContext functionality that handles this.

But would this be right, in custom app:
1. if NEXT_LOCALE is set, then trust it because automatic locale detection does that.
2.If NEXT_LOCALE is not set, then fetch req header in custom App and use that locale to determine the right currency?

Would that be a mimic?

**Reply by z4nr34l** [reply]
If you are using https://nextjs.org/docs/advanced-features/i18n-routing and set `i18n.defaultLocale` in `next.config.js` 2nd case will not happen, so if you will use
```
const router = useRouter()
const { locale, defaultLocale, pathname } = router
```
you can use `locale` const to get currently used language by next.js which should be enought for detecting default currency.
Only thing that need to be made is overriding currency for language like en-US lang and RMB currency
